api_reflector/admin.py

```diff
@@ -2,16 +2,59 @@
 Declares the flask-admin instance and sets up the model views.
 """
 
-from flask_admin import Admin
+from flask_admin import Admin, AdminIndexView
 from flask_admin.contrib.sqla import ModelView
+from flask_dance.contrib.azure import azure
+from flask import redirect, url_for
+from oauthlib.oauth2.rfc6749.errors import TokenExpiredError
 
 from api_reflector import models, db
 
-admin = Admin(name="API Reflector", template_mode="bootstrap3")
+
+class RestrictedAdminView(AdminIndexView):
+    """
+    Overrides default Flask-Admin admin view to implement OSS authentication before accessing.
+    """
+
+    def is_accessible(self):
+        if azure.authorized:
+            try:
+                resp = azure.get("/v1.0/me")
+                assert resp.ok
+                return True
+            except TokenExpiredError:
+                return redirect(url_for("azure.login"))
+        return False
+
+    def inaccessible_callback(self, name, **kwargs):
+        return redirect(url_for("azure.login"))
+
+
+admin = Admin(name="API Reflector", template_mode="bootstrap3", index_view=RestrictedAdminView())
+
+
+class RestrictedView(ModelView):
+    """
+    Overrides ModelView to implement OSS authentication before accessing.
+    """
+
+    def is_accessible(self):
+        if azure.authorized:
+            try:
+                resp = azure.get("/v1.0/me")
+                assert resp.ok
```

Comment:
Please change the asserts to a check. Asserts can be optimised out if certain flags are set.